odoo/addons/base/models/res_partner.py

```diff
@@ -508,6 +514,9 @@ def _fields_sync(self, values):
             if self.parent_id and self.type == 'contact':
                 onchange_vals = self.onchange_parent_id().get('value', {})
                 self.update_address(onchange_vals)
+        # sync salesperson with parent
+        if not self.user_id and not values.get('user_id') and values.get('parent_id') and self.company_type == 'person':
+            self.write({'user_id': self.parent_id.user_id.id})
```

Comment:
would be better to do `super(Partner, self).write({'user_id': self.parent_id.user_id.id})` like we do in `update_address` to avoid loop (write call _fields_sync which will call write again)